### PR TITLE
TIP-1227: Add ProductModel projection model

### PR DESCRIPTION
**Pull request by pierallard**

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Model/ElasticsearchProductModelProjection.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+namespace Akeneo\Pim\Enrichment\Bundle\Elasticsearch\Model;
+
+use Akeneo\Pim\Enrichment\Component\Product\Model\ProductModelInterface;
+
+/**
+ * @author    Pierre Allard <pierre.allard@akeneo.com>
+ * @copyright 2019 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ElasticsearchProductModelProjection
```
Comment (ahocquard) [inline src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Model/ElasticsearchProductModelProjection.php]: final

File: src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Model/ElasticsearchProductModelProjection.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        $this->allIncomplete = $allIncomplete;
+        $this->ancestorIds = $ancestorIds;
+        $this->ancestorCodes = $ancestorCodes;
+        $this->ancestorLabels = $ancestorLabels;
+        $this->label = $label;
+        $this->ancestorAttributeCodes = $ancestorAttributeCodes;
+        $this->attributesForThisLevel = $attributesForThisLevel;
+    }
+
+    public function toArray(): array
```
Comment (ahocquard) [inline src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Model/ElasticsearchProductModelProjection.php]: missing spec :)

File: src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Model/ElasticsearchProductModelProjection.php (hunk 1 above)
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Model/ElasticsearchProductModelProjection.php]: final

File: src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Model/ElasticsearchProductModelProjection.php (hunk 2 above)
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Model/ElasticsearchProductModelProjection.php]: spec

**Pull request merged**